beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -175,23 +164,18 @@ public void onSlot(UInt64 slot) {
     }
   }
 
-  private BlockPerformance getBlockPerformanceForEpochs(
-      UInt64 startEpochInclusive, UInt64 endEpochExclusive) {
-    int numberOfBlockProductionAttempts =
-        blockProductionAttemptsByEpoch
-            .subMap(startEpochInclusive, true, endEpochExclusive, false)
-            .values()
-            .stream()
-            .mapToInt(AtomicInteger::get)
-            .sum();
-    List<SlotAndBlockRoot> producedBlocks =
-        producedBlocksByEpoch
-            .subMap(startEpochInclusive, true, endEpochExclusive, false)
-            .values()
-            .stream()
-            .flatMap(Collection::stream)
-            .collect(Collectors.toList());
+  private BlockPerformance getBlockPerformanceForEpoch(UInt64 currentEpoch) {
+    int numberOfBlockProductionAttempts = 0;
+    AtomicInteger blocksAtEpoch = blockProductionAttemptsByEpoch.get(currentEpoch);
+    if (blocksAtEpoch != null) {
+      numberOfBlockProductionAttempts = blocksAtEpoch.get();
+    }
 
+    List<SlotAndBlockRoot> producedBlocks = new ArrayList<>();
```

Comment:
better to use `Collections.emptyList()` rather than creating a `new ArrayList<>()` instance here.